model2vec/inference/model.py

```diff
@@ -305,3 +348,46 @@ def evaluate_single_or_multi_label(
     )
 
     return report
+
+
+def get_most_important_tokens(model: Any, text: str) -> list[tuple[str, float]]:
+    """
+    Get the most important tokens for the predicted class.
+
+    :param model: The model to get the most important tokens for.
+    :param text: The text to get the most important tokens for.
+    :return: A list of (token, score) tuples sorted by descending importance.
+    """
+    # Use the model's tokenizer to convert the text to token IDs
+    if isinstance(model, StaticModelPipeline):
+        input_ids = model.model.tokenize([text])
+    else:
+        input_ids = model.tokenize([text]).tolist()
+    unique_ids = set(input_ids[0])
+
+    # Identify tokens that are not yet cached and compute their logits
+    tokens_to_compute = [token_id for token_id in unique_ids if token_id not in model.token_logits_cache]
```

Comment:
maybe you can just do this in the loop below? For each token, you just get it from the cache, if you miss, you compute it? Just a small idea